onward/app/weather/WeatherApi.scala

```diff
@@ -36,7 +38,20 @@ object WeatherApi extends ExecutionContexts with ResourcesHelper {
     if (Play.isTest) {
       Future(Json.parse(slurpOrDie(new URI(url).getPath.stripPrefix("/"))))
     } else {
-      WS.url(url).get().filter(_.status == 200).map(_.json)
+      if(conf.switches.Switches.RetryFailedAccuWeatherApiRequests.isSwitchedOn) {
+        val fetchRequest = () => WS.url(url).withRequestTimeout(300).get().filter(_.status == 200).map { response =>
```

Comment:
Can we pull out the request timeout, and the retry backoff to constant values defined at the object level. Helps with diffs if we need to modify this in the future